db/sql/filter.go

```diff
@@ -0,0 +1,176 @@
+package sql
+
+import (
+	"strconv"
+
+	sq "github.com/Masterminds/squirrel"
+	"github.com/cloudwan/gohan/db/search"
+	"github.com/cloudwan/gohan/schema"
+)
+
+const (
+	OrCondition   = "__or__"
+	AndCondition  = "__and__"
+	BoolCondition = "__bool__"
+)
+
+type queryBuilder interface {
+	Where(pred interface{}, args ...interface{})
```

Comment:
Why not `Where (pred interface{}, args ...interface{}) queryBuilder`, to be consistent with the squirrel library?

Reply:
I recall I started with this approach but for some reason I chose to remove return value. I'm not going to discover it again.